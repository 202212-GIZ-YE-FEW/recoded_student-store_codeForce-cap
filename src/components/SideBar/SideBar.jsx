import Image from "next/image"

import Highlighter from "../highlighter"

export default function SideBar({ handleSelectedPage }) {
  return (
    <>
      {/* // If it is large screen */}
      <section className='hidden lg:block shadow-2xl overflow-hidden h-[100%]'>
        <div className='flex flex-col justify-around text-center w-[331px] pt-10 h-[80%]'>
          <Image
            className='rounded-full mx-auto'
            src='/productImg.png'
            alt='...'
            width={171}
            height={171}
          />
          <div className='text-[22px]'>
            <h2 className='font-semibold'>User Name</h2>
            <p>User Email</p>
            <p>User Location</p>
            <button
              aria-label='Edit Profile'
              onClick={() => handleSelectedPage("EditProfile")}
            >
              <Highlighter highlighterStyle='editProfile' text='Edit Profile' />
            </button>
          </div>
          <div className='grid grid-rows-2 gap-8'>
            <button
              aria-label='My Listing'
              onClick={() => handleSelectedPage("UserListings")}
            >
              <Highlighter highlighterStyle='listingPage' text='My Listings' />
            </button>
            <button
              aria-label='My Orders'
              onClick={() => handleSelectedPage("UserOrders")}
            >
              <Highlighter highlighterStyle='ordersPage' text='My Orders' />
            </button>
          </div>
        </div>
      </section>
      {/* //If the expression false */}
      <section className='block lg:hidden'>
        <div className='grid grid-cols-3 text-[17px] items-center text-purple-dark text-center bg-[#90EEE1] h-[38px]'>
          <button
            className='focus:bg-[#7874F2] focus:text-[#d7d7d7]'
            aria-label='My Orders'
            onClick={() => handleSelectedPage("UserOrders")}
          >
            My Orders
          </button>
          <button
            className='focus:bg-[#7874F2] focus:text-[#d7d7d7]'
            aria-label='My Listing'
            onClick={() => handleSelectedPage("UserListings")}
          >
            My Listing
          </button>
          <button
            className='focus:bg-[#7874F2] focus:text-[#d7d7d7]'
            aria-label='Edit Profile'
            onClick={() => handleSelectedPage("EditProfile")}
          >
            Edit Profile
          </button>
        </div>
        <br />
        <br />
        <Highlighter highlighterStyle='editProfile' text='Edit Profile' />
        <hr className='w-[82%] h-[1px] mx-auto bg-gray-100 border-0 rounded dark:bg-gray-700'></hr>
      </section>
    </>
  )
}
