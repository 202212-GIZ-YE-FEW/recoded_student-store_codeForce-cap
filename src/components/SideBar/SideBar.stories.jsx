import SideBar from "./SideBar"

export default {
  title: "Components/SideBar",
  component: SideBar,
  argTypes: {
    currentPage: {
      control: {
        type: "select",
        options: ["editProfile", "myListing", "myOrders"],
      },
    },
    img: { control: { type: "text" } },
    name: { control: { type: "text" } },
    email: { control: { type: "text" } },
    location: { control: { type: "text" } },
    link: { control: { type: "text" } },
    backgroundColor: { control: { type: "color" } },
  },
}

const Template = ({ backgroundColor, ...args }) => (
  <div style={{ backgroundColor }}>
    <SideBar {...args} />
  </div>
)

export const LargeScreen = Template.bind({})
LargeScreen.args = {
  currentPage: "editProfile",
  img: "/productImg.png",
  name: "John Doe",
  email: "john.doe@example.com",
  location: "New York",
  link: "/",
  backgroundColor: "#f5f5f5",
}

export const SmallScreen = Template.bind({})
SmallScreen.args = {
  currentPage: "editProfile",
  img: "/productImg.png",
  name: "John Doe",
  email: "john.doe@example.com",
  location: "New York",
  link: "/",
  backgroundColor: "#f5f5f5",
}
