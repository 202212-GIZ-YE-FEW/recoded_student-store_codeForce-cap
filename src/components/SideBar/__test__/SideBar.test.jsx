import { render } from "@testing-library/react"

import SideBar from "../SideBar"

describe("SideBar", () => {
  it("renders large screen version correctly", () => {
    const { container } = render(
      <SideBar
        img='/productImg.png'
        name='Test Name'
        email='test@test.com'
        location='Test Location'
        link='/testLink'
        currentPage='editProfile'
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders small screen version correctly", () => {
    const { container } = render(
      <SideBar
        img='/productImg.png'
        name='Test Name'
        email='test@test.com'
        location='Test Location'
        link='/testLink'
        currentPage='otherPage'
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it("does not render the image in small screen if not on edit profile page", () => {
    const { container } = render(
      <SideBar
        img='/productImg.png'
        name='Test Name'
        email='test@test.com'
        location='Test Location'
        link='/testLink'
        currentPage='otherPage'
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
