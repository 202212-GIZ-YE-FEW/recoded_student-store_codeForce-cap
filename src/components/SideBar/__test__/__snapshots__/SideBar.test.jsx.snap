// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar does not render the image in small screen if not on edit profile page 1`] = `
<section
  class="hidden lg:block lg:w-[331px] lg:h-[947px] text-center lg:shadow-2xl lg:pt-12"
>
  <div
    class="flex flex-col justify-around h-[75%]"
  >
    <img
      alt="..."
      class="rounded-full mx-auto mb-3"
      data-nimg="1"
      decoding="async"
      height="171"
      loading="lazy"
      src="/_next/image?url=%2FproductImg.png&w=384&q=75"
      srcset="/_next/image?url=%2FproductImg.png&w=256&q=75 1x, /_next/image?url=%2FproductImg.png&w=384&q=75 2x"
      style="color: transparent;"
      width="171"
    />
    <div
      class="text-[22px]"
    >
      <h2
        class="font-semibold"
      >
        Test Name
      </h2>
      <p>
        test@test.com
      </p>
      <p>
        Test Location
      </p>
    </div>
    <button
      href=""
    >
      <span>
        <span
          class="undefined "
        >
          <h2>
            Text
          </h2>
        </span>
      </span>
    </button>
    <div
      class="font-bold text-[31px] grid grid-rows-2 gap-8"
    >
      <h1>
        <button
          aria-label="My Listing"
          href=""
        >
          My Listing
        </button>
      </h1>
      <h1>
        <button
          aria-label="My Orders"
          href=""
        >
          My Orders
        </button>
      </h1>
    </div>
  </div>
</section>
`;

exports[`SideBar renders large screen version correctly 1`] = `
<section
  class="hidden lg:block lg:w-[331px] lg:h-[947px] text-center lg:shadow-2xl lg:pt-12"
>
  <div
    class="flex flex-col justify-around h-[75%]"
  >
    <img
      alt="..."
      class="rounded-full mx-auto mb-3"
      data-nimg="1"
      decoding="async"
      height="171"
      loading="lazy"
      src="/_next/image?url=%2FproductImg.png&w=384&q=75"
      srcset="/_next/image?url=%2FproductImg.png&w=256&q=75 1x, /_next/image?url=%2FproductImg.png&w=384&q=75 2x"
      style="color: transparent;"
      width="171"
    />
    <div
      class="text-[22px]"
    >
      <h2
        class="font-semibold"
      >
        Test Name
      </h2>
      <p>
        test@test.com
      </p>
      <p>
        Test Location
      </p>
    </div>
    <button
      href=""
    >
      <span>
        <span
          class="bg-[url('/images/Highlighter.png')] bg-bottom bg-no-repeat text-purple-light font-poppins text-center font-bold tracking-[-1px] block md:bg-[length:313.92px_27.28px] md:text-[27.28px] bg-[length:233.67px_20.31px] text-[21px] lg:bg-[length:200px_38px] lg:text-[30px] "
        >
          <h2>
            Text
          </h2>
        </span>
      </span>
    </button>
    <div
      class="font-bold text-[31px] grid grid-rows-2 gap-8"
    >
      <h1>
        <button
          aria-label="My Listing"
          href=""
        >
          My Listing
        </button>
      </h1>
      <h1>
        <button
          aria-label="My Orders"
          href=""
        >
          My Orders
        </button>
      </h1>
    </div>
  </div>
</section>
`;

exports[`SideBar renders small screen version correctly 1`] = `
<section
  class="hidden lg:block lg:w-[331px] lg:h-[947px] text-center lg:shadow-2xl lg:pt-12"
>
  <div
    class="flex flex-col justify-around h-[75%]"
  >
    <img
      alt="..."
      class="rounded-full mx-auto mb-3"
      data-nimg="1"
      decoding="async"
      height="171"
      loading="lazy"
      src="/_next/image?url=%2FproductImg.png&w=384&q=75"
      srcset="/_next/image?url=%2FproductImg.png&w=256&q=75 1x, /_next/image?url=%2FproductImg.png&w=384&q=75 2x"
      style="color: transparent;"
      width="171"
    />
    <div
      class="text-[22px]"
    >
      <h2
        class="font-semibold"
      >
        Test Name
      </h2>
      <p>
        test@test.com
      </p>
      <p>
        Test Location
      </p>
    </div>
    <button
      href=""
    >
      <span>
        <span
          class="undefined "
        >
          <h2>
            Text
          </h2>
        </span>
      </span>
    </button>
    <div
      class="font-bold text-[31px] grid grid-rows-2 gap-8"
    >
      <h1>
        <button
          aria-label="My Listing"
          href=""
        >
          My Listing
        </button>
      </h1>
      <h1>
        <button
          aria-label="My Orders"
          href=""
        >
          My Orders
        </button>
      </h1>
    </div>
  </div>
</section>
`;
